### PR TITLE
Remove `Localization.getLocalizedStringWithNamespace`

### PR DESCRIPTION
**Pull request by mattbjordan**
Remove the getLocalizedStringWithNamespace method of [ITwinLocalization](https://github.com/iTwin/itwinjs-core/blob/95c4c6cf8645377b3e34debf0231d6d0e56a0429/core/i18n/src/ITwinLocalization.ts#L153) and the [Localization](https://github.com/iTwin/itwinjs-core/blob/28c52844af49df44c40d7e9c41f8a3e4915b372a/core/common/src/Localization.ts#L69) interface
 as discussed [here](https://github.com/iTwin/itwinjs-core/pull/4323#discussion_r1016939303). **This is a breaking change** - do not merge this PR.

The removal should happen _after_ the [localization tests PR](https://github.com/iTwin/itwinjs-core/pull/4323) gets merged, thus this was created as a draft PR. Once the tests are merged in, this branch can merge in master and this PR can be finished up with `extract-api` and `change` updates.

Still todo in this PR:

- [x] Wait for https://github.com/iTwin/itwinjs-core/pull/4323 to be merged
- [ ] Merge master into this branch
- [ ] Run `rush extract-api`
- [ ] Run `rush change`

### CODE REVIEW

File: core/common/src/Localization.ts [hunk 1]
```diff
@@ -60,13 +60,6 @@ export interface Localization {
    * @throws Error if no keys resolve to a string.
    */
   getLocalizedString(key: string | string[], options?: TranslationOptions): string;
-  /** Similar to `getLocalizedString` but the namespace is a separate param and the key does not include the namespace.
-   * @param namespace - the namespace that identifies the particular localization file that contains the property.
-   * @param key - the key that matches a property in the JSON localization file.
-   * @returns The string corresponding to the first key that resolves.
-   * @throws Error if no keys resolve to a string.
-   */
-  getLocalizedStringWithNamespace(namespace: string, key: string | string[], options?: TranslationOptions): string;
```
Comment (pmconne) [inline core/common/src/Localization.ts:69]: Isn't this a breaking `@public` API change?

Comment (calebmshafer) [reply core/common/src/Localization.ts:69]: gah, I didn't realize that this was public in the interface. It's technically internal in the iTwinLocalization implementation.

I agree with @pmconne, mark as deprecated in this PR is probably the way to go and filing an issue to remove in 4.0.

Comment (mattbjordan) [reply core/common/src/Localization.ts:69]: It was easier for me to deprecate in a new PR: https://github.com/iTwin/itwinjs-core/pull/4701. Would it be okay to leave this PR here as a draft for now?

Comment (pmconne) [reply core/common/src/Localization.ts:69]: Sure.

**Pull request closed without merging**